src/webcam.vue

```diff
@@ -11,66 +11,67 @@
 
 <script>
 export default {
-  name: "VueWebCam",
+  name: 'VueWebCam',
 
   props: {
     width: {
       type: [Number, String],
-      default: "100%"
+      default: '100%',
     },
     height: {
       type: [Number, String],
-      default: 500
+      default: 500,
     },
     autoplay: {
       type: Boolean,
-      default: true
+      default: true,
     },
     screenshotFormat: {
       type: String,
-      default: "image/jpeg"
+      default: 'image/jpeg',
     },
     selectFirstDevice: {
       type: Boolean,
-      default: false
+      default: false,
     },
     deviceId: {
       type: String,
-      default: null
+      default: null,
     },
     playsinline: {
       type: Boolean,
-      default: true
+      default: true,
     },
     resolution: {
       type: Object,
       default: null,
-      validator: value => {
-        return value.height && value.width;
-      }
-    }
+      validator: value => value.height && value.width,
+    },
   },
 
+  emits: ['cameras', 'camera-change', 'video-live', 'notsupported', 'started', 'stopped', 'error'],
+
   data() {
     return {
       source: null,
       canvas: null,
       camerasListEmitted: false,
-      cameras: []
+      cameras: [],
+      localDeviceId: this.deviceId,
     };
   },
 
   watch: {
-    deviceId: function(id) {
+    deviceId(id) {
       this.changeCamera(id);
-    }
+    },
   },
 
   mounted() {
     this.setupMedia();
   },
 
-  beforeDestroy() {
```

Comment:
@danaki Wouldn't this break it for vue 2?

Reply:
I guess yes